devices/YarpRobotLoggerDevice/app/robots/iCubGazeboV3/yarp-robot-logger.xml

```diff
@@ -8,6 +8,20 @@ GNU Lesser General Public License v2.1 or any later version. -->
   <param name="sampling_period_in_s">0.01</param>
   <param name="port_prefix">/yarp-robot-logger</param>
 
+  <group name="Telemetry">
+    <param name="save_period">600.0</param>
+  </group>
+
+  <group name="ExogenousSignals">
+    <param name="exogenous_inputs">("Walking")</param>
+
+    <group name="Walking">
+      <param name="port_name">"/yarp-robot-logger/exogenous_signals/walking:i"</param>
+      <param name="signal_name">"walking"</param>
+    </group>
```

Comment:
@GiulioRomualdi I had a question regarding this. Why is this an input port?
Why don't we do any connection with an external port?
Could you explain a bit more about the usage here? I am sorry I missed this detail during the review.

Reply:
I was wondering, If there's an arbitrary port existing in the network, for example `/icub/xsens_inertial/analog:o` then is it possible to log data directly from that port through this exogenous signal connection? If yes, how do we do that?

Reply:
ping @GiulioRomualdi

Reply:
Hi @prashanthr05 this is the port opened by the device this is the reason why it's an input port. The connection is left to the user. 
`/icub/xsens_inertial/analog:o` cannot be used as it is indeed the signal must be compliant with this format
 
https://github.com/ami-iit/bipedal-locomotion-framework/blob/5d1d6b81f498602ffc6c83e48e4d3c707a2e6cd5/src/YarpUtilities/thrifts/BipedalLocomotion/YarpUtilities/VectorsCollection.thrift#L3-L6

Reply:
Ah ok, so we might need a wrapper for the port `/icub/sens_inertial/analog:o`? Could you direct me to the link of how the `walking-controller` data is packed?

Reply:
Found it here: https://github.com/robotology/walking-controllers/pull/97.